sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/ExceptionNotThrown.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    class ExceptionNotThrown
+    {
+        public void Test1()
+        {
+            var e = new Exception(); // Noncompliant {{Throw this exception or remove this useless statement.}}
+            throw new Exception(); // Compliant
+        }
+
+        public Exception Test2()
+        {
+            return new Exception(); // Compliant
+        }
+
+        public Exception Test3()
+        {
+            var e = new Exception(); // Noncompliant, why not directly return?

```

Comment:
Some people prefer this style, IMHO this should produce the same result as Test2.